src/moscot/problems/base/_compound_problem.py

```diff
@@ -143,11 +150,25 @@ def _create_problems(
                 tgt_name = tgt
 
             problem = self._create_problem(src, tgt, src_mask=src_mask, tgt_mask=tgt_mask)
-            if callback is not None:
-                data = self._callback_handler(src, tgt, problem, callback=callback, **callback_kwargs)
-                kws = {**kwargs, **data}  # type: ignore[arg-type]
+            if xy_callback is not None:
+                xy_data = self._callback_handler(
```

Comment:
Can you please refactor this a bit please? Maybe allow `callback` in `_callback_handler` to be `None` and then return `{}` in this case.